internal/internal_utils.go

```diff
@@ -286,12 +290,12 @@ func getMetricsScopeForLocalActivity(ts *metrics.TaggedScope, workflowType, loca
 	return ts.GetTaggedScope(tagWorkflowType, workflowType, tagLocalActivityType, localActivityType)
 }
 
-func getTimeoutTypeFromErrReason(reason string) s.TimeoutType {
+func getTimeoutTypeFromErrReason(reason string) (s.TimeoutType, error) {
 	timeoutTypeStr := reason[strings.Index(reason, " ")+1:]
 	var timeoutType s.TimeoutType
 	if err := timeoutType.UnmarshalText([]byte(timeoutTypeStr)); err != nil {
-		// this should never happen
-		panic(err)
+		// this happens when the timeout error is constructed by an old client
```

Comment:
I'm confused. I thought timeout error is constructed by a server, not by the client.

Reply:
The timeout event is generated by the server. If a workflow gets an timeout error from an activity and return that error directly, it results in a workflow failed event. The error reason in this failed event is generated by the client side.

Reply:
I see. Would you rewrite comment to something like:
constructed by client prior version <version>